CRM/Core/DAO.php

```diff
@@ -2803,7 +2803,7 @@ public static function createSQLFilter($fieldName, $filter, $type = NULL, $alias
             // The first 2 criteria are redundant but are added as they
             // seem like they would
             // be quicker than this 3rd check.
-            && max(array_map('ord', str_split($criterion))) >= 240) {
+            && max(array_map('ord', str_split((string) $criterion))) >= 240) {
```

Comment:
so we check above
```
if (!empty($criterion) && !is_numeric($criterion)
```
because this line is really all about strings - we should probably check is_string($criterion) rather than casting a likely-array to a string here. We could probably remove that comment about the redundant too since it seems wrong once they become important checks for php8 support